### PR TITLE
map_gen: add refbox and wait pos features

### PR DESCRIPTION
**Pull request by TarikViehmann**

### CODE REVIEW

File: include/mps_map_gen/mps.hpp [hunk 1]
```diff
@@ -8,12 +8,17 @@
...
 
 namespace mps_map_gen {
+
 class MPS {
 public:
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW
   MPS(geometry_msgs::msg::TransformStamped tf, std::string name);
   MPS(Eigen::Vector2f center, Eigen::Rotation2Df rot, std::string name);
 
+  // TODO: CONFIG
```
Comment (snoato) [inline include/mps_map_gen/mps.hpp]: todo solved? or do you mean you want to put this into a config file?

File: include/mps_map_gen/mps.hpp [hunk 2]
```diff
@@ -30,4 +35,13 @@ class MPS {
   Eigen::Vector2i GetMax(float resolution);
   Eigen::Vector2i GetMin(float resolution);
 };
+struct MpsMapGenData {
+  bool needs_refresh = false;
+  int field_width = 0;
```
Comment (snoato) [inline include/mps_map_gen/mps.hpp:45]: should the default be a default field?

File: include/mps_map_gen/mps_map_gen.hpp [hunk 3]
```diff
@@ -40,18 +41,26 @@ class MpsMapGen : public rclcpp::Node {
   rclcpp::Publisher<nav_msgs::msg::OccupancyGrid>::SharedPtr map_pubsliher;
   std::shared_ptr<tf2_ros::Buffer> tf_buffer;
   std::shared_ptr<tf2_ros::TransformListener> tf_listener;
-  rclcpp::TimerBase::SharedPtr timer_;
+  std::shared_ptr<tf2_ros::StaticTransformBroadcaster> tf_broadcaster_;
 
-  std::vector<MPS> mps_list;
+  rclcpp::TimerBase::SharedPtr timer_;
 
   std::string mps_names[14] = {"M-BS",  "M-SS",  "M-DS",  "M-CS1", "M-CS2",
```
Comment (snoato) [inline include/mps_map_gen/mps_map_gen.hpp:57]: maybe can move this to config too down the line but okay for now

File: include/mps_map_gen/waiting_positions.hpp [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+public:
+  WaitPosGen(std::shared_ptr<MpsMapGenData> data);
+  std::map<std::string, Eigen::Vector3f> generate_wait_pos(int num_free_zones);
+
+private:
+  std::shared_ptr<MpsMapGenData> data_;
+  std::map<uint16_t, std::vector<Eigen::Vector2i>> zone_blocking_ = {
+      {0,
+       {
+           // 0°
```
Comment (snoato) [inline include/mps_map_gen/waiting_positions.hpp:19]: ouchie my formatter hurts

File: launch/mps_map_gen.launch.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+    # Define launch configuration variables
+    get_data_from_refbox = LaunchConfiguration('get_data_from_refbox', default=True)
+    map_client = LaunchConfiguration('map_client', default="/map_server/map")
+    publish_wait_pos = LaunchConfiguration('publish_wait_pos', default=True)
+    peer_address = LaunchConfiguration('peer_address', default="127.0.0.1")
+    recv_port_private = LaunchConfiguration('recv_port_private', default=4441)
+    recv_port_public = LaunchConfiguration('recv_port_public', default=4444)
+    crypto_key = LaunchConfiguration('crypto_key', default="randomkey")
+    # resolution = LaunchConfiguration('resolution', default=0.05)
```
Comment (snoato) [inline launch/mps_map_gen.launch.py]: why is this commented out? do we not use it?

File: src/mps_map_gen.cpp [hunk 6]
```diff
@@ -2,16 +2,64 @@
...
+  declare_parameter<std::string>(
+      "proto_path",
+      ament_index_cpp::get_package_share_directory("rcll_protobuf_msgs") +
+          "/rcll-protobuf-msgs/");
+
+  data_ = std::make_shared<MpsMapGenData>();
+  use_refbox_data_ = this->get_parameter("get_data_from_refbox").as_bool();
+  publish_wait_pos_ = this->get_parameter("publish_wait_pos").as_bool();
+  if (publish_wait_pos_) {
+    RCLCPP_INFO(get_logger(), "ACTSCHUALLY");
```
Comment (snoato) [inline src/mps_map_gen.cpp]: huhm huhm

File: src/waiting_positions.cpp [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+        [output_pos](const Eigen::Vector2i &lhs, const Eigen::Vector2i &rhs) {
+          double d_lhs = (lhs.cast<float>() - output_pos).norm();
+          double d_rhs = (rhs.cast<float>() - output_pos).norm();
+          return d_lhs < d_rhs;
+        });
+    ori_vec = (*(free_zones.begin())).cast<float>() - output_pos;
+    wait_ori = std::atan2(ori_vec.x(), ori_vec.y());
+    auto mid_point_output = zone_to_point(*(free_zones.begin()));
+    res.emplace(
+        mps.name_ + "-O-WAIT",
```
Comment (snoato) [inline src/waiting_positions.cpp]: KONSISTENZY

**Review by snoato — DISMISSED** [review]

File: include/mps_map_gen/mps_map_gen.hpp (hunk 3 above)
Comment (TarikViehmann) [reply include/mps_map_gen/mps_map_gen.hpp:57]: yea could be a future option, but these names are only used for the alternative mode for static transform publishing anyways.

File: include/mps_map_gen/mps.hpp (hunk 2 above)
Comment (TarikViehmann) [reply include/mps_map_gen/mps.hpp:45]: this is now set via ros2 parameter retrieval

File: include/mps_map_gen/mps.hpp (hunk 1 above)
Comment (TarikViehmann) [reply include/mps_map_gen/mps.hpp]: that was a useles TODO; these values are the actual measurements of the mps, no need to configure them.

**Review by techtasie — APPROVED** [review]
Looking good to me :+1: No complaints from my side!

**Comment by techtasie** [comment]
@TarikViehmann you should merge if you think it's ready

**Pull request merged**